opm/simulators/utils/DeferredLogger.cpp

```diff
@@ -82,10 +82,18 @@ namespace Opm
         messages_.push_back({Log::MessageType::Note, "", message});
     }
 
-    void DeferredLogger::logMessages()
+    void DeferredLogger::logMessages(bool convert_error_to_warning)
     {
-        for (const auto& m : messages_) {
-            OpmLog::addTaggedMessage(m.flag, m.tag, m.text);
+        if(convert_error_to_warning) {
+            for (const auto& m : messages_) {
+                OpmLog::addTaggedMessage((m.flag==Log::MessageType::Error)?
```

Comment:
Instead of converting, why we do not treat those as `Log::MessageType::Error` in the first place? That means, we change the definition/usage of OPM_DEFLOG_THROW or related a little bit?

Reply:
Because these methods are used at various different places. For some of those they are an Error.

An alternative would be to use additional special Exceptions for errors that are supposed to be ignored (and not NumericalProblem as now) and treat those as warnings in the DeferredLogger.